docs/src/filters/capture.md

```diff
@@ -8,9 +8,25 @@ This is often used as a way of retrieving authentication tokens from a packet, a
 [ConcatenateBytes](./concatenate_bytes.md) and 
 [TokenRouter](token_router.md) filter to provide common packet routing utilities.
 
+### Capture strategies
+
+There are multiple strategies for capturing bytes from the packet.
+
+#### Suffix
+Captures bytes from the end of the packet.
+
+#### Prefix
+Captures bytes from the start of the packet.
+
+#### Regex
+Captures bytes using a regular expression. Unlike other capture strategies,
+the regular expression can return one or many values if there are
+multiple matches.
```

Comment:
> Also wondering if we need to be explicit that a Regex filter that returns more than one value will be dropped by the Token Router.

Well that's why it currently returns a single value, so that it continues to work with token_router. I wonder if we enforce regex to always be a list, that we should also change token_router to check if anything in the list matches?

Reply:
Except `Regex` can return a `Value::List` or a single value?

https://github.com/googleforgames/quilkin/blob/2828b8678d6ab3440ddc61eac27e6264f5528bed/src/filters/capture/regex.rs#L20-L26

Or am I reading `matches.into_iter().next()` wrong?

> that we should also change token_router to check if anything in the list matches?

There's probably an interesting discussion of having an option on `token_router` to allow matching against lists of values - shall we create an issue for it? (seems to me like it should be opt in - optimise on the extra security/validation first).

Reply:
> Except Regex can return a Value::List or a single value?

Yes, currently but @iffyio mentioned the alternative being always have it be a list even when it's a single value. 

> (seems to me like it should be opt in - optimise on the extra security/validation first)

This is kinda why I think I'd lean more towards returning a single value over a list with a single value, so that it can be compatible with token router, since most of the you probably only want a single capture group anyway.

I'll update token router's documentation to say that it requires a single value.